apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
     for (var key in keyHandlerSet) {
       setHandlerMap(key);
     }
+
+    //add Observer for SEND_READ_REPORT_KEY
+    settings.addObserver(this.SEND_READ_REPORT_KEY, function (event) {
+      Settings.sendReadReportStatus = event.settingValue;
+      console.log(Settings.sendReadReportStatus);
+    });
+  },
+
+  sendReadReport: function toSendReadReport() {
+    var deferred = Utils.Promise.defer();
+    if(Settings.sendReadReportStatus != null) {
+      deferred.resolve(Settings.sendReadReportStatus);
+    } else {
+      deferred.reject('ERROR: Settings.sendReadReportStatus unavailable');
+    };
+
+    return deferred.promise;
+  },
+
+  //Set sendReadReportStatus for SEND_READ_REPORT_KEY
+  toSendReadReport: function toSendReadReport(status) {
+    if (!isNaN(status)) {
+      Settings.sendReadReportStatus = status;
+      console.log(Settings.sendReadReportStatus);
+    }
   },
```

Comment:
why we are deleting this._sendReadReportDefer,  here in initSendReadReport. Shouldn't it be deleted in init, once initSendReadReport is done and then after assigning this.sendReadReport = this._sendReadReportDefer.promise; then delete it.  
So, it should be like this: 
1.in init-> this._sendReadReportDefer = Utils.Promise.defer(); then,
2. function initSendReadReport is called [setHandlerMap(key) function] -> resolve promise here.
3. in init -> assigning this.sendReadReport = this._sendReadReportDefer.promise;
4 in init -> then delete, delete this._sendReadReportDefer;
Just going through execution sequence.
@julienw  please correct me if i am wrong here.